src/infra/main/template.yaml

```diff
@@ -197,6 +215,258 @@ Resources:
       AliasName: !Sub "alias/${AWS::StackName}-SQSEncryptionKey"
       TargetKeyId: !GetAtt SQSEncryptionKey.Arn
 
+  AccountDeletionProcessorFunction:
+    Type: AWS::Serverless::Function
+    # checkov:skip=CKV_AWS_116: DLQ is not appropriate for a Lambda invoked by an API
+    # checkov:skip=CKV_AWS_115: Function is not configured for function-level concurrent execution Limit
+    Properties:
+      FunctionName: !Sub '${AWS::StackName}-AccountDeletionProcessorFunction'
+      Description: "A lambda function that updates an account status indicating that the account has been deleted"
+      CodeUri: ../../../dist/handlers/account-deletion-processor-handler
+      Timeout: 60
+      Handler: account-deletion-processor-handler.handler
+      Role: !GetAtt AccountDeletionProcessorFunctionRole.Arn
+      Environment:
+        Variables:
+          TABLE_NAME:
+            Fn::ImportValue: !Sub "${CoreStackName}-AccountStatusTableName"
+          DELETED_ACCOUNT_STATUS_RETENTION_SECONDS: 63072000
+          METRIC_SERVICE_NAME: !Sub "${AWS::StackName}-AccountDeletionProcessorFunction"
+      Events:
+        SNSEvent:
+          Type: SNS
+          Properties:
+            RedrivePolicy: !Sub "{ \"deadLetterTargetArn\": \"arn:aws:sqs:${AWS::Region}:${AWS::AccountId}:${AWS::StackName}-AccountDeletionProcessorDLQ\" }"
```

Comment:
Should be able to dynamically reference the DLQ ARN as I don't see a reference from DLQ to this function. So therefore there wouldn't be a circular dependency